packages/wonder-blocks-form/src/components/text-field.js

```diff
@@ -78,39 +93,56 @@ export default class TextField extends React.Component<Props, State> {
         focused: false,
     };
 
-    handleOnChange: (event: SyntheticInputEvent<HTMLInputElement>) => mixed = (
-        event,
-    ) => {
-        const {validation, onValidation, onChange} = this.props;
-        const newValue = event.target.value;
+    componentDidMount() {
+        this.maybeValidate(this.props.value);
+    }
+
+    maybeValidate: (newValue: string) => void = (newValue) => {
+        const {validation, onValidation} = this.props;
         if (validation) {
             const maybeError = validation(newValue);
```

Comment:
Somewhat unrelated but I wonder if we should make sure the value falls back to being `null` as the default, if no response is provided, that way it'll match the initial state of `error` (right now if `undefined` is returned that will cause a re-render).
```suggestion
            const maybeError = validation(newValue) || null;
```

Reply:
Thanks on that! I was treating `undefined` and `null` as the same, but you're right, it would cause a re-render when `error` goes from its initial `null` state to an `undefined` state on the first invalid input.